issue/591/triaxiality: Build scripts in examples/triaxiality [version:1.18.0]

shenmingfu opened the pull request:
#591

shenmingfu committed:
built a folder examples/triaxiality, added a readme file inside

shenmingfu committed:
added mock catalog info into readme file

shenmingfu committed:
added mock catalogs, added a notebook for visualization

akumgill committed:
test notebook plotting delta sigma vs radius

... using equations 31-38 from Tae-hyeon's paper (https://academic.oup.com/mnras/article/475/2/2421/4788804)

rkrishnan2912 committed:
Model predictions for Quadrupole lensing

akumgill committed:
updating notebook with gamma sign and cos term sign

... also adding const plots

akumgill committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

akumgill committed:
code cleanup / removing dupe loops

rancesol committed:
put functions for data analysis into script file

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

rkrishnan2912 committed:
Comparison of two implementations

Comparison of Tae's and Adhikari's implementations of quadrupole and monopole components

rancesol committed:
adding fitting files

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

rancesol committed:
update fitting notebook

rancesol committed:
fixed emcee fitting error due to inconcistent parameter values

rancesol committed:
cleaning up repeated data files

rancesol committed:
update fitting nnotebook to account for corrected monopole term

rkrishnan2912 committed:
MCMC fitting for elliptical lens model

Includes fitting for orientation of halo and ellipticity simultaneously.

rkrishnan2912 committed:
Fixed the fitting modules

Commit:
Added triaxiality theory functions in func_layer.py

Commit:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

Commit:
some changes to initialize triaxiality functions

Commit:
Merge branch 'main' into issue/591/triaxiality

rancesol committed:
added triaxial functions to class modules

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

Update triaxiality code.

tae-h-shin committed:
added quadrupole calculation plus major axis calculation

tae-h-shin committed:
fixed conflicts

Commit:
documentation updated

rancesol committed:
add triaxial functions

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

Update data_ops

rancesol committed:
typo correction

Commit:
Minor comments and cosmetic changres

Commit:
Data operation-functions for triaxiality added to dataops

tae-h-shin committed:
edited galaxycluster.py to incorporate quadrupole lensing

tae-h-shin committed:
edited galaxycluster.py to incorporate quadrupole lensing

tae-h-shin committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

tae-h-shin committed:
changed dataops/__init__.py for quadrupole

tae-h-shin committed:
changed galaxycluser.py for quadrupole

Commit:
Edited a minor typo in code in dataops/__init__.py

Commit:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

rancesol committed:
argument and backend validation

Commit:
Edited a small typo in dataops/__init__.py

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

merge in dataops updates

tae-h-shin committed:
initial edit to clusterensemble.py for quadrupole

tae-h-shin committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

tae-h-shin committed:
finished adding quadrupole operations to clusterensemble.py

tae-h-shin committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

rancesol committed:
cleaned up triaxiality functions

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

updating

Commit:
Minor addition to doc string in func_layer.py

Commit:
Reformatting func_layer.py

tae-h-shin committed:
delted three redundant functions from dataopts

tae-h-shin committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

rancesol committed:
fixed formating

rancesol committed:
add triaxial tests

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

update

tae-h-shin committed:
cosmetic changes to make pylint happy and added one test

tae-h-shin committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

rancesol committed:
fix 'test_triaxial' arguments

rancesol committed:
correct arments for simpson integration function

Commit:
Pylint corrections to func_layer.py and __init__.py

rancesol committed:
triaxial test added for object-oriented and functional form comparison

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

update

tae-h-shin committed:
pylint is finally happy

tae-h-shin committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

Commit:
Updates to the interpolation and sampling in func_layer.py theory functions

rancesol committed:
Merge branch 'issue/591/triaxiality' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

ts485 committed:
issue/627/testskypixelcoords: New notebook to test coordinate system effects with instructions to use example data (#629)

* feat: Add coordinate_system parameter to compute_tangential_and_cross_components and _compute_lensing_angles functions

* Update documentation to account for new coordinate_system option

* feat: Add coordinate_system conversion test to test_compute_lensing_angles_flatsky function

* Add coordinate_system conversion test to test_compute_lensing_angles_astropy function and fix bug in coordinate system conversion bug in compute_lensing_angles_astropy

* Set coordinate_system option during creation of GalaxyCluster object

* adding a new notebook and data

* test

* new notebook to test coordinate systems and shear measurements

* Removed debuggin print in test_dataops.py

* editing README for the data availability

* moved tomomi's notebook to examples

* added prelude to Tomomi's notebook

* Fixed typo and set default coordinate_system to "pixel"

* Added option to choose coordinate system for generated mock catalog

* Added ellipticity coordinate system conversion tests to test_galaxycluster.py and test_mockdata.py

* Added documentation to other test cases in Tomomi's notebook.  Still need to incorporate Caio's added functionality for sky vs. pixel coordinates in CLMM

* Fixed formatting

* starting to work through and document example of lensing signal calculation of HSCY3 cluster without CLMM, but need to change colossus dependency because it is not easy to install on NERSC

* Finished baseline documentation, including changing colossus cosmology dependency in notebook to the astropy one so this is easier to run on NERSC

* minor change to clarify sections of notebook

* added kwargs to GalaxyCluster instance definition in preparation for re-installing latest CLMM with Caio's added coordinate_system kwargs

* Updated logic for lensing angle conversion between coordinate systems in _compute_lensing_angles_flatsky and updated tests

* Revert "Updated logic for lensing angle conversion between coordinate systems in _compute_lensing_angles_flatsky and updated tests"

This reverts commit 4a1faa62dd810870fee2873090bf55ea7d9b06a9.

* Added coordinate_system validation and extra tests

* Changed example notebook to document new coordinate_system option

* examples/test_coordinate.ipynb has additional minor markdown detail

* test_coordinate.ipynb now runs from top to bottom using Caio's coordinate system implementation!

* adding additional docstrings to Tomomi's from-scratch calculation of lensing from HSC data

* refactoring code into functions to make loop over cluster lenses  more readable and some renaming of variables, e.g. sum to profile_sum, to avoid name space problems

* made additional refactoring changes, but refactored functions to calculate angles between source and lens, etc. likely broke something in the background selection (now empty)

* some of the refactoring incorporated with pre-broken code

* cleaned up the loop over rbins

* In line comments for portions of calculation

* modified reverted commit to when the functions did not break, added in documentation and light refactoring from latest commit

* renamed the notebook for consistency

* docs/doc-config.ini now includes the demo_coordinate_system notebook

* feat: Add coordinate_system parameter to compute_tangential_and_cross_components and _compute_lensing_angles functions (#624)

* feat: Add coordinate_system parameter to compute_tangential_and_cross_components and _compute_lensing_angles functions

* Update documentation to account for new coordinate_system option

* feat: Add coordinate_system conversion test to test_compute_lensing_angles_flatsky function

* Add coordinate_system conversion test to test_compute_lensing_angles_astropy function and fix bug in coordinate system conversion bug in compute_lensing_angles_astropy

* Set coordinate_system option during creation of GalaxyCluster object

* Removed debuggin print in test_dataops.py

* Fixed typo and set default coordinate_system to "pixel"

* Added option to choose coordinate system for generated mock catalog

* Added ellipticity coordinate system conversion tests to test_galaxycluster.py and test_mockdata.py

* Fixed formatting

* Updated logic for lensing angle conversion between coordinate systems in _compute_lensing_angles_flatsky and updated tests

* Revert "Updated logic for lensing angle conversion between coordinate systems in _compute_lensing_angles_flatsky and updated tests"

This reverts commit 4a1faa62dd810870fee2873090bf55ea7d9b06a9.

* Added coordinate_system validation and extra tests

* Changed example notebook to document new coordinate_system option

* Fixed all the documentation for coordinate_system and changed names from 'pixel'/'sky' to 'euclidean'/'celestial'

* Added reference for defitions of coordinate systems

* Fixed bug in coordinate system conversion in mock data generation

* Test ValueError raise for coordinate_system in test_mockdata.py

* Added more explanations about ellipticity coordinate system

* Fixed typo

* uodated tag to 1.12.4

---------

Co-authored-by: Marina Ricci <rm003211@gmail.com>

* removed a character in a markdown cell that tests complained about in making docs

* swapped pixel/euclidean and sky/celestial in notebook

* reverted two files to try to match main

* removed outputs in the notebook

* removed OTHER notebook outputs

* remove empty cells in notebooks

* made minor edits in response to review, primarily with generalizing file paths and removing comments or redundant text in the markdown

* incorporating original demo_mock_cluster.ipynb from main to remove changes to the file in this branch that should not have been committed

* modified the Oguri example to also show incorrect coordinate assumption as noted by Caio

* got rid of dollar signs inside markdown code snippet env hoping this will get the checks passed

* added review contributions line :-)

* added a note for the CosmoDC2 clusters so the user is aware to not necessarily expect a strong signal in center regions

* removed extra cell at end

* removed rogue savefig

* clarified why there might be less signal in the TXPipe outputted CosmoDC2 thing

* Formatted demo_coordinate_system_datasets.ipynb notebook

---------

Co-authored-by: Caio Lima de Oliveira <caiooliveiracode@proton.me>
Co-authored-by: Camille Avestruz <briarrose5187@gmail.com>
Co-authored-by: Marina Ricci <rm003211@gmail.com>

rancesol committed:
minor edits to triaxial functs to bring into agreement with naming schemes used in clmm

rancesol committed:
edits to triaxial functs to bring into agreement with func_layer.py

rancesol committed:
added tests for triaxial functions

rancesol committed:
ran black on files

rancesol committed:
removed unused function

rancesol committed:
remove obsolete function from test

tae-h-shin committed:
finished test_dataops.py

tae-h-shin committed:
finished test_galaxycluster.py

tae-h-shin committed:
finished test_clusterensemble.py

rancesol committed:
add triaxial functions from func_layer

rancesol committed:
Merge branch 'main' into issue/591/triaxiality

rancesol committed:
raise value error if term is incorrect for triaxial calculations

rancesol committed:
minor edit

rancesol committed:
fixed local variables in triaxial calculations

rancesol committed:
fixed number of local variables in triaxial calculations

rancesol committed:
pylint ignore number of instances (8/7)

rancesol committed:
minor formating edit

combet commented:
Hi - I was looking at this branch as I was attempting to summarizing all the new function names/functionalities for the v2.0 paper. 

I understand that the main function for the prediction of the quadrupole components is in `eval_excess_surface_density_triaxial` in `parent_class.py` in the `theory` module. However, I do not see a corresponding `compute_excess_surface_density_triaxial` function in the functional interface (`func_layer.py`). 

If memory serves, the theory module was originally designed to have a “mirror” implementation in the functional interface (func.py) and in the object-oriented interface (parent_class.py). See e.g, the 2 theory demo notebooks (eval_* methods of the CLMM modeling class all have a corresponding compute_* function in the functional interface). I'd be tempted to say we should try to keep this when developing new functionalities. What do you think? @shenmingfu, @rancesol, @tae-h-shin, @rkrishnan2912,  @m-aguena, @marina-ricci

combet commented on clmm/dataops/__init__.py:
```diff
@@ -111,6 +124,20 @@ def compute_tangential_and_cross_components(
         Critical (effective) surface density in units of :math:`M_\odot\ Mpc^{-2}`.
         Used only when is_deltasigma=True.
+    include_quadrupole: bool
+        If `True`, the quadrupole shear components (g_4theta, g_const; Shin+2018) are calculated
+        instead of g_t and g_x
```
From what I see in the code, that case returns the quadrupole components _as well as_ g_t, g_x, not _instead_. Or I am misunderstanding something?

rancesol replied:
You are correct. I remember this was a last minute change -- we just forgot to edit the description. Thanks for catching that!

rancesol commented:
Hi @combet , I agree on the naming scheme between the parent_class and func_layer. We had ensured that functionally they were identical but how they are called differs. We will correct this as soon as possible.

rancesol committed:
corrected description on include_quadrupole parameter

rancesol committed:
mirrored triaxial functions in func_layer with their counterparts in parent_class

rancesol committed:
updated triaxial function imports in __init__

rancesol commented:
Hi @combet, thanks for the comments. They've both been resolved now.

combet commented on clmm/__init__.py:
```diff
@@ -24,6 +24,12 @@
     Modeling,
     Cosmology,
 )
+from .theory.func_layer import (
+    compute_delta_sigma_4theta_triaxiality,
+    compute_delta_sigma_const_triaxiality,
+    compute_delta_sigma_excess_triaxiality,
+)
+
```
@rancesol - These functions do not exist anymore after your last commit (-->  single function `compute_excess_surface_density_triaxial` instead )

rancesol replied:
Sorry, I missed that file. Should be fixed now.

rancesol committed:
updated names of triaxial functions

combet committed:
NB showing issue in quad measurement

combet commented:
Hi - I added a notebook `examples/triaxiality/clmm_triax_test_cc` showing the issue I mentioned regarding the measurement of the quadrupole with celestial coordinates. The markdown should be self-explanatory, but let me know if something is unclear. @rkrishnan2912, @shenmingfu, @rancesol, @tae-h-shin

tae-h-shin committed:
fixed the coordinate system issue (see the slack message)

rkrishnan2912 committed:
Notebook showcasing CLMM func_layer implementation

rkrishnan2912 committed:
Edited Outputs

m-aguena committed:
fix merge conflicts

rkrishnan2912 committed:
Horizontally-rescaled mock catalogs

Contails the mock catalogs with the correct horizontal re-scaling that preserves volume of the spherical cluster halo as it is stretched into an elliptical halo.

rkrishnan2912 committed:
Update test_theory.py

Resolved triaxial function name mismatches.

rkrishnan2912 committed:
CLMM triaxiality example notebook

A notebook to showcase the triaxial functions implemented in CLMM and fit Mock shear catalogs. This notebook reproduces fitting results (Fig. 12) in the CLMMv2 paper.

m-aguena commented on clmm/dataops/__init__.py:
```diff
@@ -158,15 +193,40 @@ def compute_tangential_and_cross_components(
+        if sigma_c is not None:
+            four_theta_comp *= _sigma_c_arr
+            const_comp *= _sigma_c_arr
+    if include_quadrupole:
```
this if is repeated

rkrishnan2912 committed:
Update __init__.py

Removed repeated if condition (if include_quadrupole)

rkrishnan2912 committed:
Update test_dataops.py

Editing the shears to account for change in coordinate system from Euclidean to Celestial.

rkrishnan2912 committed:
Update test_galaxycluster.py

Changing shear sign for accounting difference between Euclidean and Celestial coordinate system

rkrishnan2912 committed:
Update test_galaxycluster.py

Undoing changes to test_coordinate_system

rkrishnan2912 committed:
Update conftest.py

Updating tolerance table: CCL tolerance from 3.0E-8 to 1E-7

rkrishnan2912 committed:
Update conftest.py

Updating Tolerance Table: Relative tolerance from 3.0E-8 to 1E-7 for Numcosmo and Nobackend cases.

tae-h-shin committed:
Update test_galaxycluster.py

updated the input shear_2 in test_coordinate_system to reflect the recent coordinate system changes.

tae-h-shin committed:
Update __init__.py

The way I implemented the coordinate system changes are wrong -- I modified the coordinate system inside the code, but it should be dealt outside of the code.

tae-h-shin committed:
Update test_dataops.py

The corresponding changes to the above commit, in testing functions.

rkrishnan2912 committed:
Update func_layer.py

Pylint suggestions addressed

rkrishnan2912 committed:
Update func_layer.py

Pylint trailing whitespaces removed

rkrishnan2912 committed:
Update __init__.py

Pylint warnings addressed

rkrishnan2912 committed:
Update __init__.py

Pylint warnings addressed

rkrishnan2912 committed:
Update func_layer.py

Pylint warnings addressed

rkrishnan2912 committed:
Update __init__.py

Pylint warning for whitespaces addressed

rkrishnan2912 committed:
Update __init__.py

Docstring backticks warning resolved. (WARNING: Inline interpreted text or phrase reference start-string without end-string.)

rkrishnan2912 committed:
Update galaxycluster.py

Docutils warning addressed

rkrishnan2912 committed:
Update galaxycluster.py

Docutils warnings addressed

rkrishnan2912 committed:
Update galaxycluster.py

Docutils warnings addressed

rkrishnan2912 committed:
Update galaxycluster.py

Docutils warnings addressed

Commit:
Merge branch 'main' of https://github.com/LSSTDESC/CLMM into issue/591/triaxiality

m-aguena commented:
I am still reviewing this PR, but please add the tests for the missing lines (check [coveralls](https://coveralls.io/builds/73689739))

m-aguena committed:
simplify quadrupole use in ClusterEnsemble.make_stacked_radial_profile

m-aguena committed:
simplify quadrupole use in ClusterEnsemble.make_stacked_radial_profile

m-aguena committed:
simplify quadrupole use in ClusterEnsemble.compute_sample_covariance

rancesol committed:
add ValueError tests for triaxial functions

rancesol committed:
Add ValueError test when neight phi_major or member distributions are suplied but include_quadrupole=True

rancesol committed:
fixed typo in ValueError test for quadrupole function

rancesol committed:
corrected aguments for compute_tangential_and_cross_compenents ValueError test

rkrishnan2912 committed:
Update test_dataops.py

Adding unit tests to check the Phi_major transformation under celestial coordinate system

rkrishnan2912 committed:
Update __init__.py

Updated description for Phi_major argument in compute_tangential_and_cross_components(). When this quantity is directly provided, it is assumed to be in Euclidean coordinates exclusively. In the event of celestial coordinates, the transformation happens in the code (Phi_major -> Pi + Phi_major).

m-aguena commented on clmm/theory/parent_class.py:
```diff
@@ -270,6 +264,60 @@ def _eval_excess_surface_density_2h(
+        eta_grid = grid * np.gradient(np.log(sigma0_grid), grid)
+        eta = InterpolatedUnivariateSpline(grid, eta_grid, k=5)(r_proj)
+
+        if term == "mono":
```
I suggest creating a `triaxiality.py` file and putting them there actually

m-aguena reviewed (changes requested):
Hi guys, great work!

I left a few points to be checked.
I still have to check the notebooks

rkrishnan2912 committed:
Delete examples/triaxiality/data_functions.py

Redundant!

rancesol committed:
file no longer needed

rkrishnan2912 commented on clmm/__init__.py:
```diff
@@ -24,6 +24,10 @@
     Cosmology,
 )
+from .theory.func_layer import (
+    compute_excess_surface_density_triaxial,
```
Removed the file. (Not in use)

rkrishnan2912 committed:
Delete examples/triaxiality/readme

Not in use

rkrishnan2912 committed:
Delete examples/triaxiality/model_functions.py

Not in use

rkrishnan2912 committed:
Delete examples/triaxiality/fitting_elliptical_halo.ipynb

Not in use

rkrishnan2912 committed:
Delete examples/triaxiality/delta_sigma_test.ipynb

Not in use

rkrishnan2912 committed:
Delete examples/triaxiality/catalogs.zip

Not in use

rkrishnan2912 committed:
Delete examples/triaxiality/Elliptical_lenses_model.ipynb

Not in use

rkrishnan2912 committed:
Delete examples/triaxiality/Elliptical_lenses_MCMC.ipynb

Not in use

rkrishnan2912 committed:
Delete examples/triaxiality/DESY1_Elliptical_lenses_MCMC.ipynb

Not in use

rancesol commented on examples/triaxiality/data_functions.py:
```diff
@@ -0,0 +1,39 @@
+	inds = np.digitize(r, rbin_edges, right=True) - 1
+	rbin_mean = np.array([np.mean(r[inds==i]) for i in range(Nbins)])
+	return r, rbin_edges, rbin_mean, inds
+
```
This file is no longer relevant and has been removed.

rancesol committed:
imported compute_excess_surface_density_triaxial from theory layer instead of theory.func_layer layer

rancesol committed:
moved triaxiality computations for mono, quad_4theta, and quad_const to separate functions

rancesol committed:
Added more description on the 'term' option

shenmingfu commented on clmm/dataops/__init__.py:
```diff
@@ -106,6 +118,19 @@ def compute_tangential_and_cross_components(
+        the direction of the major axis of the input cluster in the unit of radian. 
+        only needed when `include_quadrupole` is `True`.
+        Users could choose to provide ra_mem, dec_mem and weight_mem instead of this quantity.
+    info_mem : list of array, optional
```
This is to compute the orientation angle of the major axis. Earlier this angle is computed internally. Now we allow the user to provide an angle value directly. @tae-h-shin will adjust the code.

tae-h-shin committed:
Update clusterensemble.py  (g4theta, e4theta, gconst, econst ==> g_quad_4theta etc.)

shenmingfu commented on clmm/clusterensemble.py:
```diff
@@ -170,20 +210,33 @@ def make_individual_radial_profile(
-            cross_component_out,
-            weights_out,
-        )
+        if self.include_quadrupole:
```
@m-aguena This is under the `def make_individual_radial_profile()`. If we remove this if statement, then quadruple s will be calculated every time. We would like to let the user to decide whether to compute quadruples. If only a spherical cluster model is considered, then quadruples will not be necessary (this if statement will save time and resources).

rkrishnan2912 replied:
@m-aguena Yes, you are right. This line can be removed, it is redundant as it is being checked in `add_individual_radial_profile()` that `self.include_quadrupole` is `True`. 

I have removed that line. Let us know if it looks good.

tae-h-shin committed:
Update galaxycluster.py (e4theta, econst, g4theta, gconst ==> e_quad_4theta etc.)

tae-h-shin committed:
Update test_clusterensemble.py (changed e.g. g4theta ==> g_quad_4theta etc.)

tae-h-shin committed:
Update test_galaxycluster.py (changed e.g. e4theta ==> e_quad_4theta etc.)

tae-h-shin committed:
Update test_clusterensemble.py (chagned gconst ==> g_quad_const)

rancesol committed:
cleaned with black and made _calculate_major_axis a public funciton

rancesol committed:
unused parameter estimates were removed

rancesol committed:
adjust length of docstring

shenmingfu committed:
adjusted clmm/theory/func_layer.py for indentation

m-aguena committed:
simplify clmm/clusterensemble.py

rkrishnan2912 committed:
Update clusterensemble.py

Removed the redundant condition to check if `self.include_quadrpole` is set to be `True` in `make_individual_radial_profile()` function. This is being checked in `add_individual_radial_profile()` and is unnecessary here.

m-aguena committed:
fix merge conflicts

m-aguena committed:
fix imports

m-aguena committed:
readd deleted file

rkrishnan2912 committed:
Update clusterensemble.py

Incorporated suggestions on code-formatting

rkrishnan2912 committed:
Update clusterensemble.py

Incorporated code-formatting suggestions

rkrishnan2912 committed:
Update __init__.py

Incorporated code-formatting suggestions

rkrishnan2912 committed:
Update parent_class.py

Incorporated code-formatting suggestions

rkrishnan2912 committed:
Update __init__.py

Incorporated code-formatting suggestions

rkrishnan2912 committed:
Update galaxycluster.py

rkrishnan2912 committed:
Update clusterensemble.py

rkrishnan2912 committed:
Update clusterensemble.py

rkrishnan2912 committed:
Update __init__.py

rkrishnan2912 committed:
Update __init__.py

rkrishnan2912 commented on clmm/theory/func_layer.py:
```diff
@@ -1331,3 +1324,77 @@ def compute_magnification_bias(
+
+
+def compute_excess_surface_density_triaxial(
+    r_proj, mdelta, cdelta, z_cl, ell, cosmo, term, n_grid=10000, **kwargs
```
Yes, it is passed to the compute_surface_density function for additional parameters like alpha_ein, r_mis, etc.

rkrishnan2912 commented on clmm/dataops/__init__.py:
```diff
@@ -69,6 +73,14 @@ def compute_tangential_and_cross_components(
+    are also calculated using the two ellipticity/shear components :math:`g_1` and :math:`g_2` 
+    of the source galaxies, following Eq.31 and Eq.34 in Shin et al. (2018), arXiv:1705.11167. 
+
+    .. math::
```
Yep, this looks good!

m-aguena committed:
Merge branch 'main' into issue/591/triaxiality

m-aguena committed:
Merge branch 'main' into issue/591/triaxiality

m-aguena committed:
fix merge conflicts

m-aguena committed:
Merge branch 'main' into issue/591/triaxiality

m-aguena committed:
fix merge conflicts

m-aguena committed:
create clmm/theory/triaxiality.py

m-aguena committed:
remove unused import

m-aguena committed:
update tests

m-aguena committed:
fix typo

m-aguena committed:
generelize computation of triaxiality functions

m-aguena committed:
use modeling object in compute_excess_surface_density_triaxial

m-aguena committed:
bump version

m-aguena committed:
run black

m-aguena committed:
run pylint

m-aguena committed:
simplify data module

m-aguena committed:
remove info_mem from compute_tangential_and_cross_components

m-aguena committed:
add phi_major attribute to GalaxyCluster

m-aguena committed:
add more tests

m-aguena committed:
fix tests

m-aguena committed:
simplify GalaxyCluster.compute_tangential_and_cross_components

m-aguena committed:
simplify clusterensemble

m-aguena committed:
simplify clusterensemble

m-aguena committed:
homogeize code in clusterensemble

m-aguena committed:
simplify galaxycluster.py

m-aguena committed:
minor style change

m-aguena committed:
simplify GalaxyCluster.plot_profiles

m-aguena committed:
add triaxiality to demo_theory_functionality_* notebooks

m-aguena committed:
add triaxiality to demo_dataops_functionality.ipynb

m-aguena committed:
revert examples/demo_mock_ensemble.ipynb examples/demo_boost_factors.ipynb

m-aguena committed:
loosen precision for triaxiality comparisons only

m-aguena committed:
clean calculate_major_axis

m-aguena committed:
fix typo

m-aguena committed:
make plot_profile return tuple instead of generator

m-aguena committed:
mv miscentering to its own function

m-aguena committed:
fix typo

m-aguena committed:
rm unnecessary tests

m-aguena committed:
flatten args

combet commented on clmm/theory/parent_class.py:
```diff
@@ -885,6 +902,74 @@ def eval_surface_density_2h(
+        verbose=False,
+        n_grid=10000,
+    ):
+        r"""Compute the individual terms in the quadrupole expansion of the excess surface density.
```
Use the same docstrings as the one from the functional interface, where reference to the paper, etc. was given (`r"""Compute the excess surface density lensing profile for the monopole, 4theta quadrupole,
    or constant quadrupole component given in Shin et al. 2018
    (https://doi.org/10.1093/mnras/stx3366)`)

m-aguena replied:
done!

combet commented on examples/demo_dataops_functionality.ipynb:
```diff
@@ -279,6 +283,50 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is also possible to compute the quadrupole compenents. In this case `compute_tangential_and_cross_components` returns (`theta`, `e_t`, `e_x`, `e_quad_4theta`, `e_quad_const`):"
```
Maybe also reference the relevant paper for the quadrupole calculation in the notebook

m-aguena replied:
done!

combet reviewed (commented):
My main comment on this PR is to add as much as possible the relevant reference/equation for the new triaxiality parts, either in the docstrings or in the demo notebooks. I commented on the parts I saw, where this could be useful.

m-aguena committed:
update documentation

m-aguena committed:
add equations in eval_excess_surface_density_triaxial

m-aguena committed:
fix docstrings

m-aguena committed:
add triax eqs to all theory docstrings

m-aguena committed:
update triax doc in dataops

m-aguena committed:
fix line lengh
